datafusion/common/src/config.rs

```diff
@@ -314,6 +314,85 @@ config_namespace! {
     }
 }
 
+/// When using the parquet feature,
+/// use the same default writer settings as the extern parquet.
+#[cfg(feature = "parquet")]
+mod parquet_defaults {
+    use parquet::basic::Compression;
+    use parquet::file::properties as props;
+
+    /// Default value for [`parquet::WriterProperties::data_page_size_limit`]
+    pub const DEFAULT_PAGE_SIZE: usize = props::DEFAULT_PAGE_SIZE;
+    /// Default value for [`parquet::WriterProperties::write_batch_size`]
+    pub const DEFAULT_WRITE_BATCH_SIZE: usize = props::DEFAULT_WRITE_BATCH_SIZE;
+    /// Default value for [`parquet::WriterProperties::writer_version`]
+    pub const DEFAULT_WRITER_VERSION: &str = "1.0";
+    /// Default value for [`parquet::WriterProperties::dictionary_enabled`]
+    pub const DEFAULT_DICTIONARY_ENABLED: Option<bool> =
+        Some(props::DEFAULT_DICTIONARY_ENABLED);
+    /// Default value for [`parquet::WriterProperties::dictionary_page_size_limit`]
+    pub const DEFAULT_DICTIONARY_PAGE_SIZE_LIMIT: usize =
+        props::DEFAULT_DICTIONARY_PAGE_SIZE_LIMIT;
+    /// Default value for [`parquet::WriterProperties::data_page_row_count_limit`]
+    pub const DEFAULT_DATA_PAGE_ROW_COUNT_LIMIT: usize =
+        props::DEFAULT_DATA_PAGE_ROW_COUNT_LIMIT;
+    /// Default value for [`parquet::WriterProperties::max_statistics_size`]
+    pub const DEFAULT_MAX_STATISTICS_SIZE: Option<usize> =
+        Some(props::DEFAULT_MAX_STATISTICS_SIZE);
+    /// Default value for [`parquet::WriterProperties::max_row_group_size`]
+    pub const DEFAULT_MAX_ROW_GROUP_SIZE: usize = props::DEFAULT_MAX_ROW_GROUP_SIZE;
+    /// Default value for [`parquet::WriterProperties::column_index_truncate_length`]
+    pub const DEFAULT_COLUMN_INDEX_TRUNCATE_LENGTH: Option<usize> =
+        props::DEFAULT_COLUMN_INDEX_TRUNCATE_LENGTH;
+
+    // TODO: discuss if we want datafusion to use these defaults from the extern parquet
+    // refer to https://github.com/apache/datafusion/issues/11367
+
+    #[allow(dead_code)]
+    /// Default value for [`parquet::WriterProperties::statistics_enabled`]
+    pub const DEFAULT_STATISTICS_ENABLED: Option<&str> = Some("page");
+    #[allow(dead_code)]
+    /// Default value for [`parquet::BloomFilterProperties::fpp`]
+    pub const DEFAULT_BLOOM_FILTER_FPP: Option<f64> =
+        Some(props::DEFAULT_BLOOM_FILTER_FPP);
+    #[allow(dead_code)]
+    /// Default value for [`parquet::BloomFilterProperties::ndv`]
+    pub const DEFAULT_BLOOM_FILTER_NDV: Option<u64> =
+        Some(props::DEFAULT_BLOOM_FILTER_NDV);
+
+    #[allow(dead_code)]
+    /// Default value for [`parquet::WriterProperties::compression`]
+    pub const DEFAULT_COMPRESSION: Compression = Compression::UNCOMPRESSED;
```

Comment:
Arrow-rs parquet's [default is uncompressed](https://github.com/apache/arrow-rs/blob/16915b57f066875b650c44edf55a748c6e49c709/parquet/src/file/properties.rs#L35). Whereas the [datafusion default is zstd(3)](https://github.com/apache/datafusion/blob/827d0e3a29c0ea34bafbf03f5102407bd8e9b826/datafusion/common/src/config.rs#L374).

I think the default for datafusion makes sense (to be compressed), but I'm unclear as to why the parquet (arrow-rs) default is uncompressed. Is this leftover? Or is intentional for the [ArrowWriterOptions](https://docs.rs/parquet/52.1.0/parquet/arrow/arrow_writer/struct.ArrowWriterOptions.html)?

Reply:
We discussed this is #7692 and decided that DataFusion would have parquet compression by default.  I am not sure if arrow-rs has different factors that make uncompressed by default preferable.

Reply:
Ok. Thank you.

Reply:
I think there is a tension between raw read performance (uncompressed) and storage efficiency (compressed). @tustvold  I think preferred arrow-rs to be on the read performance side of the equation